openpype/modules/python_console_interpreter/window/widgets.py

```diff
@@ -627,8 +628,45 @@ def add_tab(self, tab_name, index=None):
     def showEvent(self, event):
         self._line_check_timer.start()
         super(PythonInterpreterWidget, self).showEvent(event)
+        # First show setup
+        if self._first_show:
+            self._first_show = False
+            self._on_first_show()
+
         self._output_widget.scroll_to_bottom()
 
+    def _on_first_show(self):
+        # Change stylesheet
+        self.setStyleSheet(load_stylesheet())
+        # Check if splitter size raio is set
+        # - first store value to local variable and then unset it
+        splitter_size_ratio = self._splitter_size_ratio
+        self._splitter_size_ratio = None
+        # Skip if is not set
+        if not splitter_size_ratio:
+            return
+
+        # Skip if number of size items does not match to splitter
+        splitters_count = len(self._widgets_splitter.sizes())
+        if len(splitter_size_ratio) != splitters_count:
+            return
+
+        # Don't use absolute sizes but ratio of last stored sizes
+        ratio_sum = sum(splitter_size_ratio)
+        sizes = []
+        max_size = self._widgets_splitter.height()
+        cur_size = 0
+        ratio = max_size / ratio_sum
+        for size in splitter_size_ratio:
+            item_size = int(ratio * size)
+            cur_size += item_size
+            if cur_size > max_size:
+                item_size -= cur_size - max_size
+            if not item_size:
+                item_size = 1
```

Comment:
Are you sure this is necessary?

I believe that `QSplitter.setSizes()` is always relative sizes right?

The documentation for [`QSplitter.setSizes`](https://doc.qt.io/qt-5/qsplitter.html#setSizes) states:
> any additional/missing space is distributed amongst the widgets according to the relative weight of the sizes.

Which does bring me to "why did it misbehave in the first place"? Maybe that was only down to being triggered before the UI was shown?

Reply:
> Which does bring me to "why did it misbehave in the first place"? Maybe that was only down to being triggered before the UI was shown?

Tried and you're right, changed.